x-pack/plugins/triggers_actions_ui/public/application/components/add_message_variables.tsx

```diff
@@ -107,15 +105,130 @@ export const AddMessageVariables: React.FunctionComponent<Props> = ({
     return <></>;
   }
 
+  const ToolTipContent = ({ description, label }: { description: string; label: string }) => {
+    return (
+      <>
+        <EuiText
+          size="s"
+          style={{
+            fontWeight: euiTheme.font.weight.bold,
+          }}
+        >
+          {label}
+        </EuiText>
+        <EuiSpacer size="s" />
+        <hr />
+        <EuiSpacer size="s" />
+        <EuiText size="xs">{description}</EuiText>
+      </>
+    );
+  };
+
+  const renderOption = (
+    option: EuiSelectableOption<{ description?: string }>,
+    searchValue: string
+  ) => {
+    return (
+      <EuiFlexGroup data-test-subj={`variableMenuButton-${option.label}`}>
+        <EuiFlexItem>
+          <EuiText
+            size="s"
+            style={{
+              fontWeight: euiTheme.font.weight.bold,
+            }}
+          >
+            <EuiHighlight search={searchValue}>{option.label}</EuiHighlight>
+          </EuiText>
+          <EuiSpacer size="xs" />
+          {option.description && (
+            <EuiText size="xs" color="subdued">
+              <EuiToolTip
+                display="block"
+                position="top"
+                content={<ToolTipContent description={option.description} label={option.label} />}
+                data-test-subj={`${option.label}-tooltip`}
+              >
+                <TruncatedText text={option.description || ''} />
+              </EuiToolTip>
+            </EuiText>
+          )}
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  };
+
   return (
     <EuiPopover
       button={Button}
       isOpen={isVariablesPopoverOpen}
       closePopover={() => setIsVariablesPopoverOpen(false)}
-      panelPaddingSize="none"
-      anchorPosition="downLeft"
+      panelPaddingSize="s"
+      anchorPosition="upRight"
+      panelStyle={{ minWidth: 350 }}
     >
-      <EuiContextMenuPanel className="messageVariablesPanel" items={getMessageVariables()} />
+      <EuiSelectable
```

Comment:
Can we write a test for this? I believe this is a new feature and I don't see any tests associated with it

Reply:
we do have one under the functional testing and we have these ones `x-pack/plugins/triggers_actions_ui/public/application/components/add_message_variables.test.tsx`
am I missing something here?

Reply:
my bad I didnt see the functional test, works for me!